src/helpers/ball-data.js

```diff
@@ -0,0 +1,736 @@
+/*
+ * Each number is a 19-digit octal string, with each digit corresponding to a ball type.
+ * The first bit of each digit indicates legality, the second bit indicates hidden ability legality,
+ * and the third bit indicates whether a ball is breedable.
+ */
+
+module.exports = {
```

Comment:
Is this list up-to-date? I want to make sure the legality checker doesn't identify anything as illegal when it's actually legal.

I don't think I've updated it for awhile, so on the IRC bot there are some recent legal combos that are still identified as illegal (e.g. Cherish Ball Miltank).

If we don't want to constantly keep it up to date, we could avoid performing the check for anything in a cherish ball, since we're going to be validating based on an event list later on anyway.